tests/unit/test_app.py

```diff
@@ -77,6 +78,49 @@ def name(name):
     return demo
 
 
+def test_mixed_content_types():
+    demo = app.Chalice('demo-app')
+
+    with pytest.raises(ValueError):
+        @demo.route('/index', content_types=['application/octet-stream',
+                                         'text/plain'])
```

Comment:
Yeah I'd say for now, just make sure the tests your adding pass flake8.  I'd prefer not to see a huge diff of flake8 fixups in this PR and just handle it as a separate PR.